src/url-loader/multi-url-resolver.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {UrlResolver} from './url-resolver';
+
+/**
+ * Resolves a URL using multiple resolvers.
+ */
+export class MultiUrlResolver implements UrlResolver {
+  constructor(private resolvers: Array<UrlResolver>) {
+    if (!this.resolvers) {
+      this.resolvers = [];
+    }
+  }
+
+  canResolve(url: string): boolean {
+    return this.resolvers.some((resolver) => {
+      return resolver.canResolve(url);
+    });
+  }
+
+  resolve(url: string): string {
+    for (let i = 0; i < this.resolvers.length; i++) {
+      let resolver = this.resolvers[i];
+      if (resolver.canResolve(url)) {
+        return resolver.resolve(url);
+      }
+    }
+  }
```

Comment:
Fair point, our reference example package-url-resolver throws if its given a url to resolve that canResolve returned false for:

```
  canResolve(url: string): boolean {
    let urlObject = parseUrl(url);
    let pathname =
        pathlib.normalize(decodeURIComponent(urlObject.pathname || ''));
    return this._isValid(urlObject, pathname);
  }

  _isValid(urlObject: Url, pathname: string) {
    return (urlObject.hostname === this.hostname || !urlObject.hostname) &&
        !pathname.startsWith('../../');
  }

  resolve(url: string): string {
    let urlObject = parseUrl(url);
    let pathname =
        pathlib.normalize(decodeURIComponent(urlObject.pathname || ''));

    if (!this._isValid(urlObject, pathname)) {
      throw new Error(`Invalid URL ${url}`);
    }
```

I feel like we should add a comment to url-resolver.ts about the expected behavior of resolve() when provided an unresolveable url.